libtransmission/CMakeLists.txt

```diff
@@ -2,6 +2,16 @@ project(libtr)
 
 configure_file(version.h.in version.h)
 
+set(THIRD_PARTY_SOURCES ConvertUTF.c jsonsl.c wildmat.c)
+if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+  set(DISABLE_WARNINGS -w)
+elseif(CMAKE_C_COMPILER_ID STREQUAL "Clang")
+  set(DISABLE_WARNINGS -Wno-everything)
```

Comment:
Not sure about this one. It counters `-Weverything` (if it was specified anywhere) which should include every warning there is, but does seem to depend on its position in command line (if some other warning flag was specified) while `-w` doesn't, e.g.

```console
$ cat a.c
int main()
{
    int x;
    return 0;
}
$ clang -Wall a.c
a.c:3:9: warning: unused variable 'x' [-Wunused-variable]
    int x;
        ^
1 warning generated.
$ clang -Weverything a.c
a.c:3:9: warning: unused variable 'x' [-Wunused-variable]
    int x;
        ^
1 warning generated.
$ clang -Wall -Wno-everything a.c
$ clang -Wno-everything -Wall a.c
a.c:3:9: warning: unused variable 'x' [-Wunused-variable]
    int x;
        ^
1 warning generated.
$ clang -Wall -w a.c
$ clang -w -Wall a.c
$ clang -Weverything -w a.c
$ clang -w -Weverything a.c
```

Reply:
Agreed. using `-w` in Clang looks like the better option. Changed 71f197423c38468d8cee82ced957e64daa776bad